packages/amino/src/encoding.ts

```diff
@@ -30,6 +32,49 @@ const pubkeyAminoPrefixSr25519 = fromHex("0dfb1005" + "20" /* fixed length */);
 /** See https://github.com/tendermint/tendermint/commit/38b401657e4ad7a7eeb3c30a3cbf512037df3740 */
 const pubkeyAminoPrefixMultisigThreshold = fromHex("22c1f7e2" /* variable length not included */);
 
+/**
+ * Decodes a multisig pubkey to type object.
+ * Pubkey structure [ prefix + const + threshold + loop:(const + pubkeyLength + pubkey            ) ]
+ *                  [   4b   + 1b    +   1b      + loop:(1b    +    1b        + pubkeyLength bytes) ]
+ * @param data encoded pubkey
+ */
+function decodeMultisigPubkey(data: Uint8Array): MultisigThresholdPubkey {
+  if (data.length < 4 + 1 + 1) { // verify that we can read at least threshold
+    throw new Error("Invalid multisig data length.");
+  }
+  let rest = data.slice(pubkeyAminoPrefixSecp256k1.length);
+  rest = rest.slice(1); // removing 0x08;
+  const threshold = rest[0];
+  rest = rest.slice(1); // removing threshold
```

Comment:
This is not constant 1 byte but a variable length int. Could you create a decoder similar to `encodeUvarint` for clarity? It is okay if ot only supports values up to 127

Reply:
ups, I expected  uint8 here, but you are right - it is varint!